rholang/src/main/scala/coop/rchain/rholang/interpreter/normalize.scala

```diff
@@ -147,7 +148,10 @@ object CollectionNormalizeMatcher {
           case tm: TupleMultiple => Seq(tm.proc_) ++ tm.listproc_.asScala.toList
         }
         foldMatch(input.knownFree, ps.toList, ETuple.apply)
-      case cs: CollectSet => foldMatch(input.knownFree, cs.listproc_.asScala.toList, ESet.apply)
+      case cs: CollectSet =>
+        val constructor: (Seq[Par], BitSet, Boolean) => ParSet =
+          (pars, locallyFree, connectiveUsed) => ParSet(SortedHashSet(pars), connectiveUsed)
```

Comment:
Also here, I am not passing `locallyFree` explicitly but trust that computing them from `Par`s that go into the `ParSet` will yield the same result. Looking at `foldMatch` implementation it looks like it will.

Reply:
That's correct. Doing it lazily should be fine.